### PR TITLE
Decaf

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR targets the machine decaf branch available for this repo.

It's worth to say that because this repo has no built in tools to decaf, we couldn't rely on those. And instead used [`decaffeinate-project.org`](https://decaffeinate-project.org/repl) to handle our decaffeination for us.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "autocomplete-atom-api",
   "version": "0.10.7",
   "description": "Pulsar API autocompletions",
-  "main": "./lib/main",
```
Comment (sertonix) [inline package.json:5]: Is this something I could/should do on my cleanup and rename pull requests?

File: update.js [hunk 2]
```diff
@@ -38,13 +38,13 @@ request(requestOptions, function(error, response, release) {
...
   };
 
-  return request(apiRequestOptions, function(error, response, atomApi) {
+  request(apiRequestOptions, function(error, response, atomApi) {
     if (error != null) {
       console.error(error.message);
       return process.exit(1);
     }
 
-    const {classes} = atomApi;
```
Comment (sertonix) [inline update.js:47]: This is not the same!

File: update.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ request(requestOptions, function(error, response, release) {
     return process.exit(1);
   }
 
-  const [apiAsset] = Array.from(release.assets.filter(({name}) => name === 'atom-api.json'));
```
Comment (sertonix) [inline update.js:29]: This is not the same!

File: lib/provider.js [hunk 4]
```diff
@@ -1,137 +1,158 @@
...
 const path = require('path');
 
 const CLASSES = require('../completions.json');
 
 const propertyPrefixPattern = /(?:^|\[|\(|,|=|:|\s)\s*(atom\.(?:[a-zA-Z]+\.?){0,2})$/;
 
-module.exports = {
-  selector: '.source.coffee, .source.js',
-  filterSuggestions: true,
+let packageDirectories = [];
```
Comment (sertonix) [inline lib/provider.js:9]: Moving to a style without `this` prevents overwriting `packageDirectories` directly and therefor would be a lost of hackability. Same thing for all other variables and functions that are exported by the module.

**Comment by confused-Techie** [comment]
@Sertonix appreciate you giving this one a review, I'll go ahead and get to most of those changes as soon as I can.

One concern here, is the recent revelation that the tester isn't working how we would hope, which means it isn't telling us if this causes failures.

So I'll get to them but we will have to wait on merging until we can get the tester working.

**Comment by confused-Techie** [comment]
Alright, I've let this one sit far to long. Plus being mixed with decaf work that concerns other contributors, and the outdated version of our tester, I'll instead go ahead and bundle this repo into the core editor, and we can do the decaf work there.

**Pull request closed without merging**